AI/AICastingComponent.cs

```diff
@@ -70,8 +69,13 @@ private void RunSpellcastingLogic()
             }
 
             Spell spell = spellBook.GetRandomSpell();
-            MagicMissionLogic.CurrentMana.TryGetValue(Agent, out var currentMana);
-            if (spell != null && currentMana >= spell.Cost && spell.IsAICastValid(Agent))
+            MagicMissionLogic.AgentsMana.TryGetValue(Agent, out var agentMana);
+            if (
+                spell != null
+                && agentMana != null
+                && agentMana.CurrentMana >= spell.Cost
+                && spell.IsAICastValid(Agent)
+            )
```

Comment:
What are your thoughts about making few helper methods for getting/setting agent manas, instead of directly accessing the `AgentsMana` property?

E.g. we could change this part into something like this
```suggestion
            if (
                spell != null
                && MagicMissionLogic.CurrentManaFor(Agent) >= spell.Cost  // If the agent has no mana component it could just return 0
                && spell.IsAICastValid(Agent)
            )
```